interceptor/src/stats/interceptor.rs

```diff
@@ -1,58 +1,217 @@
 use std::collections::HashMap;
 use std::fmt;
+use std::hash::Hash;
 use std::sync::Arc;
+use std::time::SystemTime;
 
-use super::{RTPStats, RTPStatsReader};
+use super::{StatsContainer, StatsSnapshot, StreamStats};
 use async_trait::async_trait;
-use util::sync::Mutex;
+use rtcp::payload_feedbacks::full_intra_request::FullIntraRequest;
+use rtcp::payload_feedbacks::picture_loss_indication::PictureLossIndication;
+use rtcp::receiver_report::ReceiverReport;
+use rtcp::transport_feedbacks::transport_layer_nack::TransportLayerNack;
+use rtp::extension::abs_send_time_extension::unix2ntp;
+use tokio::sync::{mpsc, oneshot};
+use tokio::time::Duration;
+
+use util::sync::{Mutex, RwLock};
 use util::{MarshalSize, Unmarshal};
 
 use crate::error::Result;
 use crate::stream_info::StreamInfo;
 use crate::{Attributes, Interceptor, RTCPReader, RTCPWriter, RTPReader, RTPWriter};
 
 #[derive(Debug)]
+enum Message {
+    StatUpdate {
+        ssrc: u32,
+        update: StatsUpdate,
+    },
+    RequestSnapshot {
+        ssrcs: Vec<u32>,
+        chan: oneshot::Sender<Vec<Option<StatsSnapshot>>>,
+    },
+}
+
+#[derive(Debug)]
+enum StatsUpdate {
+    RecvRTP {
+        packets: u64,
+        header_bytes: u64,
+        payload_bytes: u64,
+        last_packet_timestamp: SystemTime,
+    },
+    WriteRTP {
+        packets: u64,
+        header_bytes: u64,
+        payload_bytes: u64,
+        last_packet_timestamp: SystemTime,
```

Comment:
I think `Instant` as it has been used in the stats API is dubious. Instant is about measuring absolute elapsed time, but this is about a specific point in time. Ultimately this will be provided to the consumer as unix epoch seconds so `SystemTime` feels appropriate. There's a [hack](https://github.com/webrtc-rs/webrtc/blob/59be92b2dfd5400d56a3d32e6f0ed62d7e8dfd7c/webrtc/src/stats/serialize.rs#L12-L26) to convert an `Instant` to `SystemTime` which we currently use, but that feels dubious.

Reply:
Alright!